### PR TITLE
Support for safe mode preview

### PR DESCRIPTION
**Pull request by kamils-iRonin**
after merging https://github.com/theforeman/foreman/pull/8039, template rendering is broken and the error is `ArgumentError: unknown keyword: :renderer`

### CODE REVIEW

**Review by laugmanuel — APPROVED** [review]

**Comment by kamils-iRonin** [comment]
fixes #36

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,18 +30,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        foreman-core-branch: [1.23-stable, 1.24-stable, 2.0-stable, 2.1-stable, 2.2-stable, develop]
-        ruby-version: [2.3, 2.5, 2.6]
+        foreman-core-branch: [2.3-stable, develop]
```
Comment (timogoebel) [inline .github/workflows/ci.yml:33]: Do we need to bump the required version in engine.rb?

**Comment by laugmanuel** [comment]
@timogoebel, I guess we can and should merge this PR and release a new version for Foreman 2.3 as https://github.com/theforeman/foreman/pull/8039 is part of that release and the plugin is therefore broken atm.

The current implementation would mean, that the new Plugin release is incompatible with 2.2 and older.

**Review by timogoebel — APPROVED** [review]
Looks good!
@laugmanuel: Can you take care of the release?

**Comment by laugmanuel** [comment]
Thanks @kamils-iRonin !

**Comment by laugmanuel** [comment]
> @laugmanuel: Can you take care of the release?
Sure - are we ready for a new major version (1.0.0) as this is a breaking change for older Foreman versions?

**Comment by timogoebel** [comment]
> are we ready for a new major version (1.0.0) as this is a breaking change for older Foreman versions?

Yes, definitely!

**Comment by laugmanuel** [comment]
I've released foreman_vault 1.0.0 and opened a PR for packaging: https://github.com/theforeman/foreman-packaging/pull/6178
This will be cherry-picked into `rpm/2.3` after merge.

**Pull request merged**